relayer-cli/src/commands/tx/client.rs

```diff
@@ -208,7 +208,18 @@ impl Runnable for TxUpgradeClientCmd {
         let client = ForeignClient::find(src_chain, dst_chain, &self.client_id)
             .unwrap_or_else(exit_with_unrecoverable_error);
 
-        let outcome = client.upgrade();
+        // Assumption: this query is run while the chain is halted as a result of an upgrade
+        let src_upgrade_height = {
+            let src_application_height = match client.src_chain().query_latest_height() {
+                Ok(height) => height,
+                Err(e) => Output::error(format!("{}", e)).exit(),
+            };
+
+            // When the chain is halted, the application height reports a height
+            // 1 less than the halted height
+            src_application_height.increment()
+        };
```

Comment:
I guess this will go away once we have an `upgrade_height` flag? Will it be optional? In this case this code should probably go in the relayer foreign_client `upgrade()` and just pass an option there.

Reply:
Right, I am currently of the opinion that the cli should make the `upgrade_height` flag mandatory, as the operation really isn't a "give me the latest height", it's "give me the chain halt height" (otherwise it fails). We're only dealing with this "application height vs tendermint height in the case of an upgrade, and iff the user ran the command at the right time" mess because we allow the user to run the command "at just the right time", which is a bad design IMO. With the `upgrade_height` flag, all this complexity goes away.

Reply:
ok, let's leave it until #2300.